### ISSUE TITLE
Feature Request: Add support for progress indicator

### ISSUE THREAD

**Issue by JC-comp** [abraunegg/onedrive#2512]
### Is your feature request related to a problem? Please describe.

At each stage in every sync process (sync to local, consistency and integrity checks, uploading new items), it's difficult to assess the current progress without verbose logging mode, i.e. when should I expect it to be completed? if the process is actually still running?

### Describe the solution you'd like

Periodic records show current progress, including the number of items being processed, the number of items remaining (if applicable), and the time remaining (if applicable). 
Or a query interface can be called on demand.

### Describe alternatives you've considered

Detailed verbose logging would help, but it's not suitable for normal use cases. We don’t want the logs to be filled with debugging messages.

**Comment by abraunegg** [comment]
@JC-comp 
There are 3 x modes of operation - standard mode and verbose mode ... then debug.

standard mode was written to be as minimalistic as possible in output. When downloading the /delta, it does print processing '.' items to indicate it is doing *something*

I was looking to also implement that same '.' output when additional processing is occurring to be consistent - I just have not got around to it as yet.

These moving dots should then give you the visual cue that *something* is occurring.

In verbose mode (just one `--verbose`) you *do* get output detailing what it is doing - what change set, what item bundle of how many - thus you do get this level of detail already without going into debug mode.

Additionally, functional items that are in v2.4.x that still need to be added back into v2.5.x for example include the whole 'display processing metrics' function - which I think will then give you want you are looking for in a standard minimal mode + that output to tell you where things are at.

**Comment by JC-comp** [comment]
> When downloading the /delta, it does print processing '.' items to indicate it is doing something

In v2.4, I don't see any points being logged, that's why I say not knowing if the process is actually still running

And currently we can set some flags to have the following in standard mode
1. config options
2. statistic result (start/end time, # of items being processed)
3. current stage we are in
4. actions (download/upload/delete ...)

However, none of this helps in understanding the _progress_, even if the moving dots do exist, before the statistic result is logged, we are not able to know how many items we are going to deal with and how many items are left, which is hard to assess when should I expect the current sync loop to complete.

**Comment by abraunegg** [comment]
@JC-comp 
No development is going to be done with v2.4.x code.

> However, none of this helps in understanding the progress, even if the moving dots do exist, before the statistic result is logged, we are not able to know how many items we are going to deal with and how many items are left, which is hard to assess when should I expect the current sync loop to complete.

Unfortunatly there is *zero* API call to ask the API how many items in this /delta call  before you make the /delta call.

To do what you are thinking would be to make a double call - thus fetching the data twice - one to count, one to proceaa.

**Comment by JC-comp** [comment]
You are right that we have no way of knowing the total number of items for /delta call using API, but we know other operations like files to be downloaded/modifiedm. (As for total number of items, As for the total number of pieces, we can check it through the website)
And the point is, readable statistics showing when, how many files are being processed, and how many are remaining (if applicable) can help the user evaluate when the sync loop is going to complete. That's the point of this feature request.
As for v2.5, not knowing what is the final design of output for standard mode, but if moving dots is used, it can only reflect how fast the process is taking place and still not be able to evaluate the progress of each operation.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
